src/format/strftime.rs

```diff
@@ -387,12 +335,36 @@ impl<'a> StrftimeItems<'a> {
                     'U' => num0!(WeekFromSun),
                     'V' => num0!(IsoWeek),
                     'W' => num0!(WeekFromMon),
-                    'X' => queue_from_slice!(self.t_fmt),
+                    #[cfg(not(feature = "unstable-locales"))]
+                    'X' => queue_from_slice!(T_FMT),
+                    #[cfg(feature = "unstable-locales")]
+                    'X' => {
+                        if let Some(locale) = self.locale {
```

Comment:
Can you add a helper function for this? The duplication is a bit unfortunate.

Reply:
I am not great with macro's but gave it a try with a method.